research/gaia/gremlin/gremlin_core/src/process/traversal/step/source.rs

```diff
@@ -169,15 +169,9 @@ pub fn graph_step_from(
                 let mut step = GraphVertexStep::new(return_type, requirements);
                 step.set_tags(gremlin_step.get_tags());
                 let mut ids = vec![];
-                for id in opt.ids {
-                    let id = if let Ok(uid) = ID::from_str(&id) {
-                        uid
-                    } else if let Ok(id) = i64::from_str(&id) {
-                        // in case that source id is negative
-                        id as ID
-                    } else {
-                        return Err("Parse source id failed")?;
-                    };
+                for id_bytes in opt.ids {
+                    let id = read_be_u128(&mut id_bytes.as_slice());
+                    debug!("source id_bytes: {:?}, id: {:?}", id_bytes, id);
```

Comment:
Will this line produce a lot of logs when the graph is huge?

Reply:
debug!() will only print the log when the log level is debug and I think the default log level is info? Besides, here it only prints the ids in query request, not the query results.